R/get_source_expressions.R

```diff
@@ -77,6 +87,36 @@ get_source_expressions <- function(filename, lines = NULL) {
     # an error that does not use R_ParseErrorMsg
     if (is.na(message_info$line)) {
 
+      if (grepl("invalid multibyte character in parser at line", e$message, fixed = TRUE)) {
```

Comment:
One way to increase robustness is what we've done in `data.table`:

https://github.com/Rdatatable/data.table/blob/ed72e398df76a0fcfd134a4ad92356690e4210ea/inst/tests/tests.Rraw#L106-L140

Basically, keep a database (maybe populated at `onAttach()`) of error messages as generated by R itself.

That way we don't need to change anything if R updates its own message.

Just a thought, no need to do so in this PR.

Reply:
Nice idea. That would also allow us to remove the `set_lang()` construct currently in use to force english error messages.